padl/transforms.py

```diff
@@ -2191,28 +2433,84 @@ class _ItemGetter:
 
     :param samples: An object implementing __getitem__ and __len__.
     :param transform: Preprocessing transform.
+    :param entire_transform: :class:`Transform` which *transform* belongs to, i.e.,
+        :class:`Transform` whose preprocessing part is *transform*.
     :param exception: Exception to catch for (fall back to *default*).
     :param default: The default value to fall back to in case of exception.
     """
 
-    def __init__(self, samples, transform, exception=None, default=None):
+    def __init__(self, samples, transform, entire_transform, exception=None, default=None):
         self.samples = samples
         self.transform = transform
+        self.entire_transform = entire_transform
+        self.exception = exception
+        self.default = default
+
+    def __getitem__(self, item):
+        if self.exception is not None:
+            try:
+                return item, self.transform(self.samples[item])
+            except self.exception:
+                return item, self.default
+        try:
+            return item, self.transform(self.samples[item])
+        except Exception as err:
+            is_entire_transform = isinstance(self.entire_transform.pd_forward, Identity) and \
+                                  isinstance(self.entire_transform.pd_postprocess, Identity)
+            self.entire_transform._pd_trace_error(
+                self.entire_transform.pd_preprocess._pd_get_target_stage_idx(is_entire_transform),
+                [self.samples[item]]
+            )
+            raise err
+
+    def __len__(self):
+        return len(self.samples)
+
+
+class _SimpleGetter:
+    """A simple item getter.
+
+    :param samples: An object implementing __getitem__ and __len__.
+    :param exception: Exception to catch for (fall back to *default*).
+    :param default: The default value to fall back to in case of exception.
+    """
+
+    def __init__(self, samples, exception=None, default=None):
```

Comment:
@eramdiaz I think we can remove the `exception` and `default` parameters here and simplify the `__getitem__` function. We only use `_SimpleGetter` internally and the `exception` and `default` parameters are never used. If we need them in the future we can add it back, but I can't think of a use case for them at the moment.

Reply:
I agree

Reply:
Do you agree @wuhu ?

Reply:
agree!